packages/backend-core/src/events/processors/async/DocumentUpdateProcessor.ts

```diff
@@ -0,0 +1,44 @@
+import { EventProcessor } from "../types"
+import { Event, Identity, DocUpdateEvent } from "@budibase/types"
+import { DocumentType, SEPARATOR } from "../../../constants"
+import { doInTenant } from "../../../context"
+import { getDocumentId } from "../../documentId"
+import { shutdown } from "../../asyncEvents"
+
+export type Processor = (update: DocUpdateEvent) => Promise<void>
+export type ProcessorMap = { types: DocumentType[]; processor: Processor }[]
+
+export default class DocumentUpdateProcessor implements EventProcessor {
+  processors: ProcessorMap = []
+
+  constructor(processors: ProcessorMap) {
+    this.processors = processors
+  }
+
+  async processEvent(
+    event: Event,
+    identity: Identity,
+    properties: any,
+    timestamp?: string | number
+  ) {
+    const tenantId = identity.tenantId
+    const docId = getDocumentId(event, properties)
```

Comment:
Could this potentially trigger a lot of redundant user syncs? 
e.g. if a user is created we would send:
- `user:created`
- `user:builder:assigned`
- `user:admin:assigned`
- `role:assigned`

The same for any update and onboarding completed. It might be simpler / more lightweight if we add a function to `EventProcessor`: 
```ts
supports(event: Event)
```
The `DocumentUpdateProcessor` can return `true` for everything. 
Then the sub `processors` e.g. `UserSyncProcessor` can have:

```ts

const supported  = [
  Event.USER_CREATED,  
  Event.USER_UPDATED, 
  Event.USER_DELETED, 
  Event.GROUP_CREATED, 
  Event.GROUP_UPDATED,
  Event.GROUP_DELETED,
]

supports(event: Event) {
   return supported.includes(event)
}
```

It then becomes a lot easier / more specific for when we run the user sync, rather than relying on the presence of a `userId` or `groupId` field in the event properties. 

As we add more event types to the async event processing we may run into more unintended consequences of events with a user or group field running the sync. This problem can only become more compounded over time.